pkg/aws/elasticache.go

```diff
@@ -0,0 +1,137 @@
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to Get resources for clusterID : %s", clusterId)
+	}
+
+	for _, resourceTagMapping := range resourceOutput.ResourceTagMappingList {
+
+		arn := aws.StringValue(resourceTagMapping.ResourceARN)
+		arnSplit := strings.Split(arn, ":")
+		cacheClusterId := arnSplit[len(arnSplit)-1]
+		cacheClusterInput := &elasticache.DescribeCacheClustersInput{
+			CacheClusterId: aws.String(cacheClusterId),
+		}
+		cacheClusterOutput, err := r.elasticacheClient.DescribeCacheClusters(cacheClusterInput)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot get cacheclusterOutput for : %s", cacheClusterInput)
+		}
+		for _, cacheCluster := range cacheClusterOutput.CacheClusters {
+			dbLogger := logger.WithField("replicationGroup", aws.String(*cacheCluster.ReplicationGroupId))
+			if contains(replicationGroupsToDelete, *cacheCluster.ReplicationGroupId) {
+				dbLogger.Debugf("Replication Group already exists in deletion list (%s=%s)", *cacheCluster.ReplicationGroupId, clusterId)
+				break
+			}
+			replicationGroupsToDelete = append(replicationGroupsToDelete, *cacheCluster.ReplicationGroupId)
+		}
+	}
+
+	logger.Debugf("filtering complete, %d databases matched", len(replicationGroupsToDelete))
+
+	for _, replicationGroupId := range replicationGroupsToDelete {
+		//delete each replication group in the list
+
+		dbLogger := logger.WithField("replicationGroupId", aws.String(replicationGroupId))
+		dbLogger.Debugf("building report for database")
+		reportItem := &clusterservice.ReportItem{
+			ID:           replicationGroupId,
+			Name:         "elasticache ReplicationGroup",
+			Action:       clusterservice.ActionDelete,
+			ActionStatus: clusterservice.ActionStatusEmpty,
+		}
+		reportItems = append(reportItems, reportItem)
+		if dryRun {
+			dbLogger.Debug("dry run enabled, skipping deletion step")
+			reportItem.ActionStatus = clusterservice.ActionStatusDryRun
+			continue
+		}
+		dbLogger.Debug("performing deletion of database")
+		replicationGroupDescribeInput := &elasticache.DescribeReplicationGroupsInput{
+			ReplicationGroupId: &replicationGroupId,
+		}
+		replicationGroup, err := r.elasticacheClient.DescribeReplicationGroups(replicationGroupDescribeInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot describe replicationGroups")
+		}
+		//deleting will return an error if the database is already in a deleting state
+		if len(replicationGroup.ReplicationGroups) > 0 &&
+			aws.StringValue(replicationGroup.ReplicationGroups[0].Status) == statusDeleting {
+			dbLogger.Debugf("deletion of database already in progress")
+			reportItem.ActionStatus = clusterservice.ActionStatusInProgress
+			continue
+		}
+
+		deleteReplicationGroupInput := &elasticache.DeleteReplicationGroupInput{
+			ReplicationGroupId:   aws.String(replicationGroupId),
+			RetainPrimaryCluster: aws.Bool(false),
+		}
+		_, err = r.elasticacheClient.DeleteReplicationGroup(deleteReplicationGroupInput)
```

Comment:
As we're ignoring the first return value here, we could use the following syntax:

```
if _, err := ...; err != nil {
  ...
}
```

See [1] for an example.

[1] https://github.com/integr8ly/cluster-service/blob/95cd3421a83db0e95dfdfd00db9afe5c2d42fbb3/pkg/aws/testutil_test.go#L87-L90